### ISSUE TITLE
Allow changing constant display types in decompiler view

### ISSUE THREAD

**Issue by pedrib** [NationalSecurityAgency/ghidra#1752]
**Is your feature request related to a problem? Please describe.**
In IDA when you right click a constant in the decompiler view you can choose to display it as an hex int, base 10 int, char etc. This is similar to what can be done in Ghidra by going to the disassembly view, right click and select Convert. However this feature is not available in the decompiler view.

**Describe the solution you'd like**
It would be nice to mimic IDA's behaviour and allow changing the constant display type directly in the decompiler view.

**Comment by ryanmkurtz** [comment]
Related to #221

**Comment by madushan1000** [comment]
Is there any progress on this? 
When there are calculated constants in the decompiler view, it's impossible to convert them to other bases. Even if I convert every constant in the disassembly view to decimal, docompiler view will still show the calculated constants in HEX. I'm not sure it's a bug, but having the ability to convert in the decompiler view will mitigate it.

**Comment by larkwiot** [comment]
Could still use this feature, it's the number one complaint I hear from IDA users.

**Comment by fridtjof** [comment]
This goes hand in hand unfortunately with incorrect decompiler output on MIPS :/

Take this snippet for example:

```
        940ea8fc 3c 1e 94 12     lui        s8,0x9412
        940ea900 27 de 11 0c     addiu      s8,s8,0x110c
        940ea904 24 0e 00 40     li         t6,0x40
        940ea908 8f d8 00 3c     lw         t8,0x3c(s8)=>UINT_94121148                       = 3h
        940ea90c 24 0f 00 0c     li         t7,0xc
        940ea910 01 c0 00 13     mtlo       t6
        940ea914 00 00 00 00     nop
        940ea918 00 00 00 00     nop
        940ea91c 73 0f 00 00     madd       t8,t7
        940ea920 00 00 b0 12     mflo       s6
        940ea924 03 d6 20 21     addu       a0,s8,s6
```

$a0 will be the first parameter to this call:

```
iVar2 = loadlib(UINT_94121148 * 0xc + -0x6bedeeb4, ...);
```

Given that in the first snippet, an unsigned addition (addu) is performed, the decompiler should never have assumed 0x9412110C to be a signed integer. That constant is (of course) an address, and the correct output should look more like `loadlib(&libcso_proc + UINT_94121148 * 0xc, ...)` (there's a symbol at that exact address!).

Without being able to change that constant's sign, I can't even work around that misdecompilation :(

**Comment by mumbel** [comment]
> an unsigned addition (addu) is performed, the decompiler should never have assumed 0x9412110C to be a signed integer  

This is a false assumption.
`The term “unsigned” in the instruction name is a misnomer; this operation is 32-bit modulo arithmetic that does not trap on overflow. This instruction is appropriate for unsigned arithmetic, such as address arithmetic, or integer arithmetic environments that ignore overflow, such as C language arithmetic.`

**Comment by fridtjof** [comment]
Ah, sorry for the confusion. Thanks for clearing that up :)

**Comment by caheckman** [comment]
Converting the display format of integers in the Decompiler window is available starting with Ghidra 10.0.1
